daphne/src/testing.rs

```diff
@@ -470,32 +474,126 @@ impl DapLeader<BearerToken> for MockAggregator {
     }
 
-    async fn init_collect_job(&self, _collect_req: &CollectReq) -> Result<Url, DapError> {
-        unreachable!("not implemented");
+    // Called after receiving a CollectReq from Collector.
+    async fn init_collect_job(&self, collect_req: &CollectReq) -> Result<Url, DapError> {
+        let mut rng = thread_rng();
+        let task_config = self
+            .get_task_config_for(&collect_req.task_id)
+            .ok_or_else(|| DapError::fatal("task not found"))?;
+
+        let mut leader_state_store_mutex_guard = self
+            .leader_state_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+        let leader_state_store = leader_state_store_mutex_guard.deref_mut();
+
+        // Store Collect ID and CollectReq into LeaderState.
+        let collect_id = Id(rng.gen());
+        let leader_state = leader_state_store
+            .entry(collect_req.task_id.clone())
+            .or_insert_with(LeaderState::new);
+        leader_state.collect_ids.push_back(collect_id.clone());
+        let collect_job_state = CollectJobState::Pending(collect_req.clone());
+        leader_state
+            .collect_jobs
+            .insert(collect_id.clone(), collect_job_state);
+
+        // Construct a new Collect URI for this CollectReq.
+        let collect_uri = task_config
+            .leader_url
+            .join(&format!(
+                "/collect/task/{}/req/{}",
+                collect_req.task_id.to_base64url(),
+                collect_id.to_base64url(),
+            ))
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+
+        Ok(collect_uri)
     }
 
+    // Called to retrieve completed CollectResp at the request of Collector.
     async fn poll_collect_job(
         &self,
-        _task_id: &Id,
-        _collect_id: &Id,
+        task_id: &Id,
+        collect_id: &Id,
     ) -> Result<DapCollectJob, DapError> {
-        unreachable!("not implemented");
+        let mut leader_state_store_mutex_guard = self
+            .leader_state_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+        let leader_state_store = leader_state_store_mutex_guard.deref_mut();
+
+        let leader_state = leader_state_store
+            .get(task_id)
+            .ok_or_else(|| DapError::fatal("collect job not found for task_id"))?;
+        if let Some(collect_job_state) = leader_state.collect_jobs.get(collect_id) {
+            match collect_job_state {
+                CollectJobState::Pending(_) => Ok(DapCollectJob::Pending),
+                CollectJobState::Processed(resp) => Ok(DapCollectJob::Done(resp.clone())),
+            }
+        } else {
+            Ok(DapCollectJob::Unknown)
+        }
     }
 
+    // Called to retrieve pending CollectReq.
     async fn get_pending_collect_jobs(
         &self,
-        _task_id: &Id,
+        task_id: &Id,
     ) -> Result<Vec<(Id, CollectReq)>, DapError> {
```

Comment:
Somewhat orthogonal to this PR: Do you think the return type should be `HashMap<Id, CollectReq>` instead of `Vec<(Id, CollectReq)>`, in order to match `get_reports()`?

Feel free to make this change in this PR if you concur. Otherwise, I will consider doing it later.

Reply:
I've replaced `Vec<(Id, CollectReq)>` with `HashMap<Id, CollectReq>`.
Let me know what you think!